cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -223,6 +228,10 @@ public void addOutput(Output output) {
                                              "Add the output as a DatasetOutput.");
       }
       providedOutput = new ProvidedOutput(output, outputFormatProvider);
+    } else if (output.getClass().getCanonicalName().startsWith(CDAP_PACKAGE_PREFIX)) {
+      // Skip unsupported outputs from within CDAP packages.
+      // This is used to ignore unsupported outputs in MapReduce (such as the SQL Engine Output for Spark).
+      LOG.info("Unsupported output in MapReduce: {}", output.getClass().getCanonicalName());
```

Comment:
Should this be a Warning instead of an info message?

Reply:
Not really. It is perfectly valid for a plugin to register outputs that are not supported in MapReduce pipelines but are supported in Spark. 

For the BQ Sink, for example, we register one Hadoop output and one SQL Engine Output. Depending on the pipeline structure, one or the other could be used.

Reply:
@tivv I just walked through the code and tested a bit. Unfortunately, the logic to `unwrap` a wrapped collection requires a `pull` operation in order to route records to the right output port (which we use for Wrangler or other transforms with multiple potential outputs).. We need to keep this logic within the `wrappedSQLEngineCollection` class in order to prevent records from getting pulled. unless we need them to.